private/buf/cmd/buf/command/push/push.go

```diff
@@ -139,6 +150,16 @@ func run(
 	if len(flags.Tags) > 0 && flags.Draft != "" {
 		return appcmd.NewInvalidArgumentErrorf("--%s (-%s) and --%s cannot be used together.", tagFlagName, tagFlagShortName, draftFlagName)
 	}
+	if flags.CreateVisibility != "" {
+		if !flags.Create {
+			return appcmd.NewInvalidArgumentErrorf("Cannot set --%s without --%s.", createVisibilityFlagName, createFlagName)
+		}
+		// We re-parse below as needed, but do not return an appcmd.NewInvalidArgumentError below as
+		// we expect valiation to be handled here.
+		if _, err := bufcli.VisibilityFlagToVisibility(flags.CreateVisibility); err != nil {
```

Comment:
Ugh, one extra thing - sorry: we should make sure the documentation says this flag is required if --create is set, and validate that it is non-empty immediately before this line, so that there's a proper error message explaining that --create-visibility is required if --create is set. We could also take the tact to default to private, but likely better to start as required.

Reply:
Oh, I think you meant making it required when you say "force setting", in which case we wouldn't be in agreement, lol